src/main/java/org/opensearch/jdbc/ConnectionImpl.java

```diff
@@ -84,7 +85,13 @@ log, new SQLNonTransientException("Could not initialize transport for the connec
             this.clusterMetadata = connectionResponse.getClusterMetadata();
             this.open = true;
         } catch (ResponseException | IOException ex) {
-            logAndThrowSQLException(log, new SQLException("Connection error "+ex.getMessage(), ex));
+            if (ex instanceof HttpException && ((HttpException) ex).getStatusCode() == 401) {
```

Comment:
is `HttpException` an extension of IOException or ResponseException?

Reply:
ResponseException

Reply:
You can do another `catch` block for `HttpException` before `ResponseException`.

Reply:
Changed in https://github.com/Bit-Quill/sql-jdbc/pull/7/commits/2b562496734779a1c641d9a91f09454036076252